jerry-core/ecma/base/ecma-globals.h

```diff
@@ -115,6 +120,26 @@ typedef uint32_t ecma_value_t;
 #define ECMA_VALUE_SHIFT 3
 
 /**
+ * Flag for simple values
+ */
+#define ECMA_SIMPLE_VALUE_FLAG 0x8u
```

Comment:
This macro is used both as a mask and as a flag/type indicator. I'd propose splitting this into something like
```c
#define ECMA_DIRECT_VALUE_MASK 0x8u
#define ECMA_DIRECT_VALUE_INTEGER_NUMBER (0u << ECMA_VALUE_SHIFT)
#define ECMA_DIRECT_VALUE_SIMPLE_VALUE (1u << ECMA_VALUE_SHIFT)
```
And then code like `ecma_is_value_integer_number` could become:
```c
return (value & (ECMA_DIRECT_VALUE_MASK | ECMA_VALUE_TYPE_MASK)) == (ECMA_DIRECT_VALUE_INTEGER_NUMBER | ECMA_TYPE_DIRECT);
```
This yields the same binary but the source is more readable IMHO. And this could be more future-proof should any other direct values be used in addition to integers and simple values.